src/controller.cpp

```diff
@@ -57,7 +57,8 @@ extern "C" uint32_t read_controller() {
     if (mm_visible || prac_visible || settings_visible || cheats_visible ||
         tools_visible || inventory_visible || item_wheel_visible || pause_visible ||
         memory_visible || warping_visible || flags_menu_visible ||
-        scene_menu_visible || any_saves_visible || hundo_saves_visible) {
+        scene_menu_visible || any_saves_visible || hundo_saves_visible || general_flags_visible ||
+		 dungeon_flags_visible || portal_flags_visible) {
```

Comment:
This is getting a bit long, can you please abstract it out into a single function call
`static bool is_menu_open() {}` or something similar.